src/lib/arch/aarch64/Architecture.cc

```diff
@@ -235,6 +240,11 @@ std::vector<RegisterFileStructure> Architecture::getRegisterFileStructures()
 
 uint16_t Architecture::getSystemRegisterTag(uint16_t reg) const {
   assert(systemRegisterMap_.count(reg) && "unhandled system register");
+  // Check below is done for speculative instructions that may be passed into
```

Comment:
So this is to avoid spec insns that read or write to sysregs with a garbage sysreg value to not break simulation or ?

Reply:
correct, yes - just prevents garbage instructions breaking anything